giskard_vision/core/detectors/metadata_scan_detector.py

```diff
@@ -0,0 +1,164 @@
+from typing import Any, Callable, List, Sequence
+
+import numpy as np
+import pandas as pd
+
+from giskard_vision.core.detectors.base import (
+    DetectorVisionBase,
+    IssueGroup,
+    ScanResult,
+)
+from giskard_vision.utils.errors import GiskardImportError
+
+
+class MetaDataScanDetector(DetectorVisionBase):
+    """
+    Detector based on Giskard scan that looks for issues based on metadata
+
+    Args:
+        surrogate_function: function
+            Function to transform the output of the model and the ground truth into one value
+            that will be used by the scan
+        metric: function
+            Metric to evaluate the prediction with respect to the ground truth
+        type_task: str
+            Type of the task for the scan, ["regression", "classification"]
+    """
+
+    issue_group = IssueGroup(name="Metadata", description="Slices are found based on metadata")
+
+    def __init__(self, surrogate_function: Callable, metric: Callable, type_task: str) -> None:
```

Comment:
Please pass a `MetricBase` type to metric, instead of any callable, 
- you can use metric.definition in case of surrogates
- you can use metric.get in case of non-surrogates (like classification)
- metric.__name__ should be automatically handled in that case